src/service/comparison_vs_peers/comparison_vs_peers_service.py

```diff
@@ -94,11 +97,17 @@ def get_comparison_vs_data(self, data: dict, access: bool) -> list:
 
         for id in data:
             company_data = data[id]
+            company_revenue = company_data.get("actuals_revenue")
+            rule_of_40_revenue = (
+                int(company_revenue)
+                if isinstance(company_revenue, (int, float, Decimal))
+                else company_revenue
+            )
             self.calculate_metrics(company_data)
-            rule_of_40.append(self.get_rule_of_40(company_data))
-            self.remove_base_metrics(company_data)
             if not access and id not in allowed_companies:
                 self.anonymized_company(company_data, allowed_companies)
+            rule_of_40.append(self.get_rule_of_40(company_data, rule_of_40_revenue))
```

Comment:
I think we need to add a validation when rule_of_40 dict is valid, e.g. if I have revenue, but ebitda_margin is "NA" that record should be skipped.
So please, add a validation before append rule of 40, something like  is "NA' in dict.values()